### DISCUSSION TITLE
F2, F4, F5 would need a 'Magnet'

### DISCUSSION THREAD

**Discussion by flowerhorse** [brunoherbelin/vimix#82] [Ideas]
It's nearly impossible to move a user defined frame (Red with grips) manually to  'reset' Position.
I think there should be a kind of 'Magnet' That 'Snaps' the layer toHome Position or hor./vertical frame line
(similar as alpha adjusting from the 'mini' source manu)

**Comment by brunoherbelin** [comment]
For now, what is implemented is 'snap to grid' with the ALT key.
This works for move, scale and rotation.

This is not really ideal, I know,  but I also want to avoid the automatic snap when getting close to the borders: vimix never knows when the user wants to move by 1 or 2 pixels next of the border or wants to snap...

One idea is to allow creating 'grids' or 'aligment lines' that the geometry can snap to (e.g. red lines, vertical or horizontal). Would that be useful?


NB: there are also reset options in object's menus:

![Screenshot from 2023-04-12 23-03-17](https://user-images.githubusercontent.com/60393189/231584821-3005aecf-ed9a-4846-bb6e-ab415c478629.png)

![Screenshot from 2023-04-12 23-04-33](https://user-images.githubusercontent.com/60393189/231584833-4dd2122d-7646-4dcd-9d9a-0ef8ffb3cabe.png)

**Comment by flowerhorse** [comment]
> NB: there are also reset options in object's menus:

I've seen and often used them. I find them a bit cumbersome because they are first opened in a small window. The Alt Key is already faster..

**Comment by brunoherbelin** [comment]
Quite some time since this request but I can now say it is implemented in vimix 0.8.1 (now in flatpak and snap(edge)).

This works by selecting the 'Grid' dynamic mouse pointers 
![Screenshot from 2023-09-19 19-17-56](https://github.com/brunoherbelin/vimix/assets/60393189/ea1711e7-ff2b-4e58-bc41-11b6d0faf5b5)

The mouse pointers is then activated by maintaining the ALT key, 

![image](https://github.com/brunoherbelin/vimix/assets/60393189/194c9a84-9ffd-4da1-8910-3472c26cab6e)

The grid is then applied for the current action (move, rescale, rotate) on the source. Different scales of grid can be changed online with the scrollwheel of the mouse.
